zulipterminal/model.py

```diff
@@ -960,6 +961,43 @@ def get_other_subscribers_in_stream(
                 if sub != self.user_id
             ]
 
+    def _clean_custom_profile_data(
+        self, custom_profile_data: Dict[str, Dict[str, str]]
+    ) -> Dict[str, str]:
```

Comment:
As per another comment, this model function would be concerned with cleaning the data, not necessarily making it into just an output form. For example, if I want a list of users, it would be simpler to just end up with a list of ids - and this function may become simpler :)

That may mean defining additional TypedDicts for use with TidiedUserInfo and this function would be helpful.

Reply:
Should I keep this function and just use it to clean the data, and maybe create another function where I process it to get display information? 
Like the example that you gave here - when this function cleans the `list of persons` field, it then produces a list of ids in it's place. Then, a second function would fetch the names of the users with the user ids in the list, and create a string (concatenate the names together with commas) for display. 
Did I understand you correctly?

Reply:
Yes, that's what I meant. There is a function which already does this - to some extent - in UserInfoView.